### PR TITLE
fix [route] rejected output for bang input with float arguments

### PR DESCRIPTION
**Pull request by porres**
currently, nothing is output, this fixes it

closes https://github.com/pure-data/pure-data/issues/1678

### CODE REVIEW

**Comment by porres** [comment]
@umlaeute would you consider this as a fix for the development branch?

File: src/x_connective.c [hunk 1]
```diff
@@ -615,6 +616,8 @@ static void route_list(t_route *x, t_symbol *sel, int argc, t_atom *argv)
         }
     }
  rejected:
+    if(argc == 0)
+        outlet_bang(x->x_rejectout);
```
Comment (Spacechild1) [inline src/x_connective.c:620]: what is this doing?

**Comment by Spacechild1** [comment]
From the PR description I don't really understand what problem it tries to solve. Can you share a minimal patch? (screenshot is ok)

**Comment by porres** [comment]
sending a bang into [route] with float arguments should send a bang to the rightmost rejected outet, but nothing happens...

Screen Shot 2022-05-30 at 18 46 59

**Comment by Spacechild1** [comment]
I see, thanks! From the description I thought you meant `[route float]` :-)

File: src/x_connective.c (hunk 1 above)
Comment (Spacechild1) [reply src/x_connective.c:620]: I guess this is a remainder of another PR. `outlet_list` below will already output the list resp. bang.

**Comment by porres** [comment]
> From the description I thought you meant [route float] :-)

Oh, I see that now... well, seems like a simple bug fix, hopefully this is considered a no brainer and included in the development branch

**Comment by Spacechild1** [comment]
Also seems like a no-brainer to me, expect for [this line](https://github.com/pure-data/pure-data/pull/1680/files#r885087374)

**Comment by porres** [comment]
> expect for [this line](https://github.com/pure-data/pure-data/pull/1680/files#r885087374)

what about it?

File: src/x_connective.c (hunk 1 above)
Comment (Spacechild1) [reply src/x_connective.c:620]: ping @porres

Comment (porres) [reply src/x_connective.c]: how about now?

Comment (Spacechild1) [reply src/x_connective.c]: looks good now :-)

File: src/x_connective.c [hunk 2]
```diff
@@ -538,11 +538,9 @@ static void route_list(t_route *x, t_symbol *sel, int argc, t_atom *argv)
...
     if (x->x_type == A_FLOAT)
     {
-        t_float f;
-        if (!argc) return;
-        if (argv->a_type != A_FLOAT)
-            goto rejected;
-        f = atom_getfloat(argv);
+        if (!argc || argv->a_type != A_FLOAT)
+            goto rejected;;
+        t_float f = atom_getfloat(argv);
```
Comment (Spacechild1) [inline src/x_connective.c:543]: nitpick: C89 mandates that variables are declared at the top of each block.

**Pull request merged**